docs/conf.py

```diff
@@ -58,3 +54,18 @@
 extensions += ['sphinx.ext.extlinks']
 
 # local
+
+nitpick_ignore += [
+    ('py:class', 'FilesSpec'),
+    ('py:class', 'jaraco.path._ConcatenablePurePathLike'),
+    ('py:class', '_SizedT'),
+    ('py:class', 'itertools.chain'),
+    ('py:class', 'FileDescriptorOrPath'),
+    ('py:class', 'GenericPath '),
+    ('py:class', 'ReadableBuffer'),
+    ('py:class', 'StrOrBytesPath'),
+    ('py:class', 'StrPath'),
+    ('py:class', 'SupportsRead'),
+    ('py:class', 'Unused'),
+    ('py:class', '_StrOrBytesPathT'),
```

Comment:
This will potentially be fixed in Sphinx 8.2.0: https://github.com/sphinx-doc/sphinx/issues/7630